File: MsgTools/struct/java/EncodeStructmap.java
		
        if(@paramName@==null||@paramName@.size()==0){
			outputStream.writeInt(0);
		}else{
			outputStream.writeInt(@paramName@.size());
		}
        
		if(@paramName@!=null&&@paramName@.size()>0){
			for(Entry<String, @typeName@> entry:@paramName@.entrySet()){
				outputStream.writeUTF(entry.getKey());
				@valueMapEncode@;
			}
		}

File: MsgTools/struct/lua/EncodeStructmap.lua
		
		self.@paramName@ = self.@paramName@ or {}
		local @paramName@size = 0
		for k,v in pairs(self.@paramName@) do
			@paramName@size = @paramName@size + 1
		end
		outputStream:WriteInt(@paramName@size)
		for k,v in pairs(self.@paramName@) do
			outputStream:WriteUTFString(k)
			@valueMapEncode@
		end